test/index.ts

```diff
@@ -0,0 +1,202 @@
+import { expect } from 'chai';
+import {
+	boolVar,
+	checkInt,
+	HostPort,
+	intVar,
+	optionalVar,
+	requiredVar,
+	hostPortsVar,
+	trustProxyVar,
+} from '../src';
+
+const varName = 'test';
+const clearVar = () => {
+	delete process.env[varName];
+};
+const setVar = (v: string) => {
+	process.env[varName] = v;
+};
+
+beforeEach(() => {
+	clearVar();
+});
+
+describe('requiredVar', () => {
+	it('should throw when the env var is missing', () => {
+		expect(() => requiredVar(varName)).to.throw;
+	});
+	it('should return the value when the env var exists', () => {
+		const value = 'apple';
+		setVar(value);
+		expect(requiredVar(varName)).to.equal(value);
+	});
+});
+
+describe('optionalVar', () => {
+	const defaultValue = 'banana';
+	it('should return undefined when the env var is missing', () => {
+		expect(optionalVar(varName)).to.be.undefined;
+	});
+	it('should return undefined when the env var is empty', () => {
+		expect(optionalVar(varName)).to.be.undefined;
+	});
+	it('should return the default value when the env var is missing', () => {
+		expect(optionalVar(varName, defaultValue)).to.equal(defaultValue);
+	});
+	it('should return the default value when the env var is empty', () => {
+		setVar('');
+		expect(optionalVar(varName, defaultValue)).to.equal(defaultValue);
+	});
+	it('should return the value when the env var exists', () => {
+		const value = 'coconut';
+		setVar(value);
+		expect(optionalVar(varName)).to.equal(value);
+		expect(optionalVar(varName, defaultValue)).to.equal(value);
+	});
+});
+
+describe('intVar', () => {
+	const defaultValue = 'banana';
+	it('should throw when the env var is missing', () => {
+		expect(() => intVar(varName)).to.throw;
+	});
+	it('should throw when the env var is invalid', () => {
+		setVar('date');
+		expect(() => intVar(varName)).to.throw;
+	});
+	it('should return the default value when the env var is missing', () => {
+		expect(intVar(varName, defaultValue)).to.equal(defaultValue);
+	});
+	it('should return the default value when the env var is empty', () => {
+		setVar('');
+		expect(intVar(varName, defaultValue)).to.equal(defaultValue);
+	});
+	it('should return the value when the env var is a valid number', () => {
+		const value = 123;
+		setVar(`${value}`);
+		expect(intVar(varName)).to.equal(value);
+		expect(intVar(varName, defaultValue)).to.equal(value);
+	});
+});
+
+describe('boolVar', () => {
+	const defaultValue = 'banana';
+	it('should throw when the env var is missing', () => {
+		expect(() => boolVar(varName)).to.throw;
+	});
+	it('should throw when the env var is invalid', () => {
+		setVar('date');
+		expect(() => boolVar(varName)).to.throw;
+	});
+	it('should return the default value when the env var is missing', () => {
+		expect(boolVar(varName, defaultValue)).to.equal(defaultValue);
+	});
+	it('should return the default value when the env var is empty', () => {
+		setVar('');
+		expect(boolVar(varName, defaultValue)).to.equal(defaultValue);
```

Comment:
This throws as defaultValue is 'banana' so not boolean

Reply:
No it doesn't as the default value doesn't have to be a boolean, it's typed as
```ts
export function boolVar(varName: string): boolean;
export function boolVar<R>(varName: string, defaultValue: R): boolean | R;
export function boolVar<R>(varName: string, defaultValue?: R): boolean | R {
```
so if you pass in a boolean as the default then the return type will be `boolean` but if you pass in eg a string it'll be `boolean | string` and you'll be expected to handle it (I mean, you did choose to pass in a different type as the default..)